beamer/util.py

```diff
@@ -162,8 +163,8 @@ def make_web3(
     chain_id = ChainId(w3.eth.chain_id)
 
     # Apply type 2 transaction middleware for ETH2 PoS chains
-    # Currently Mainnet and Goerli
-    if chain_id in [1, 5]:
+    chain_descriptor = get_chain_descriptor(chain_id)
+    if chain_descriptor is not None and chain_descriptor.type2 and not chain_descriptor.local:
```

Comment:
Why the `not chain_descriptor.local` part?

Reply:
because we are not supporting base fee for local ?

Reply:
This is because we're enforcing Istanbul fork for local ganache, right?

Reply:
yes and if we ever allow it all fee calculations will break